django/db/models/query.py

```diff
@@ -1153,6 +1172,36 @@ def __init__(self, *args, **kwargs):
         raise TypeError("EmptyQuerySet can't be instantiated")
 
 
+class RestrictedQuerySet(object):
+    # TODO: Ideas on how to make that nicer?
+    _allowed_methods = set()
```

Comment:
Yeah probably, I posted a summary of my current questions on the ML: https://groups.google.com/forum/#!topic/django-developers/RhD-nbCqIkE -- as you noted above, keeping it a `QuerySet` might indeed be a good idea. On the other hand, `RawQuerySet` does not inherit `QuerySet` either. That probably makes sense though, since `RawQuerySet` supports nothing anymore, you can use a combined queryset in a `Paginator` at least.